src/Features/CSharp/Portable/Completion/KeywordRecommenders/ElseKeywordRecommender.cs

```diff
@@ -23,35 +23,21 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
 
             var token = context.TargetToken;
 
-            var statement = token.GetAncestor<StatementSyntax>();
-            var ifStatement = statement.GetAncestorOrThis<IfStatementSyntax>();
-
-            if (statement == null || ifStatement == null)
+            // We have to consider all ancestor if statements of the last token until we find a match for this 'else':
+            // while (true)
+            //     if (true)
+            //         while (true)
+            //             if (true)
+            //                 Console.WriteLine();
+            //             else
+            //                 Console.WriteLine();
+            //     $$
+            foreach (var ifStatement in token.GetAncestors<IfStatementSyntax>())
             {
-                return false;
-            }
-
-            // cases:
-            //   if (goo)
-            //     Console.WriteLine();
-            //   |
-            //   if (goo)
-            //     Console.WriteLine();
-            //   e|
-            if (token.IsKind(SyntaxKind.SemicolonToken) && ifStatement.Statement.GetLastToken(includeSkipped: true) == token)
-            {
-                return true;
-            }
-
-            // if (goo) {
-            //     Console.WriteLine();
-            //   } |
-            //   if (goo) {
-            //     Console.WriteLine();
-            //   } e|
-            if (token.IsKind(SyntaxKind.CloseBraceToken) && ifStatement.Statement is BlockSyntax && token == ((BlockSyntax)ifStatement.Statement).CloseBraceToken)
-            {
-                return true;
+                if (ifStatement.Statement.GetLastToken(includeSkipped: true, includeZeroWidth: true) == token)
```

Comment:
I actually have no idea why `includeSkipped: true` is needed here.

Reply:
IncludeSkipped is likely appropriate, though harder to test for.  It basically means, "if the parser  ran into stuff it totally didn't understand, but it attached it to that staement, then treat hte *last* token of that skipped run of tokens as the last token of the statement.

Reply:
What this actually does is in cases like this:
```c#
if (true)
    hello();,
```
it *prevents* `else` from being offered because context.TargetToken does not include skipped tokens.
Is that the intended behavior here?

Reply:
It seems really unlikely that there would be a skipped semicolon at the end. That usually parsers as a separate empty statement.

Reply:
In the original implementation, `else` is *not* offered in code like this (because the last token is not a semicolon or a close brace):
```c#
if (true)
    hello();,
```
But with this PR, `else` *is* offered. Which is the expected behavior here?

Reply:
> Which is the expected behavior here? 

The expected behavior is whatever is easiest to provide :)

You could literally have this work, and that would be ok.  Or it could not work, and that would be ok.

Reply:
> whatever is easiest to provide

The difference is just in setting `includeSkipped` to either true or false. Then I guess easiest to provide would be to not specify the argument?

Reply:
Given that there is absolutely no completion for anything after such skipped tokens and the fact that if you type `else` at that point it doesn't actually end up being an else clause (it also becomes a skipped token), I'm removing this.